GalilSup/src/GalilController.cpp

```diff
@@ -2903,15 +2905,18 @@ asynStatus GalilController::writeReadController(const char *caller)
   	try	{
 		//If connected, then do the command
 		if (gco_ != NULL)
-		       	{
+		{
 		 	//send the command, and get the response
 			strncpy(resp_, gco_->command(strcmd).c_str(), sizeof(resp_));
 			resp_[sizeof(resp_)-1] = '\0';
 			//No exception = success
 			done = true;
-		 	consecutive_timeouts_ = 0;
-			status = asynSuccess;
+			// Only track consecutive timeouts for same messages as in catch
+			if (strncmp(cmd_, "BP",2)==0 || strncmp(cmd_, "MT",2) == 0) {
+				consecutive_timeouts_ = 0;
 			}
```

Comment:
I see the issue is that consecutive_timeouts was being incremented by acquireDataRecord and then being reset here,  so failure was never spotted. However now consecutive_timeouts is not really “consecutive_timeouts”, maybe we need a separate timeouts record in the data records loop? Given that consecutive_timeouts will hardly ever now be reset in this loop, maybe it should be reset in acquireDataRecord when a read is successful there instead?

Reply:
Hopefully I've understood your suggestion correctly. I've split the timeout into two independent counters for each type of operation.

I've retested the solution on ALF and it works as before on sending a DR0 command.